internal/state/statuses.go

```diff
@@ -163,3 +128,75 @@ func buildStatuses(graph *graph.Graph) Statuses {
 
 	return statuses
 }
+
+func buildGatewayStatuses(
+	gateway *graph.Gateway,
+	ignoredGateways map[types.NamespacedName]*v1beta1.Gateway,
+) GatewayStatuses {
+	statuses := make(GatewayStatuses)
+
+	if gateway != nil {
+		statuses[client.ObjectKeyFromObject(gateway.Source)] = buildGatewayStatus(gateway)
+	}
+
+	for nsname, gw := range ignoredGateways {
+		statuses[nsname] = GatewayStatus{
+			Conditions:         []conditions.Condition{conditions.NewGatewayConflict()},
+			ObservedGeneration: gw.Generation,
+		}
+	}
+
+	return statuses
+}
+
+func buildGatewayStatus(gateway *graph.Gateway) GatewayStatus {
+	listenerStatuses := make(map[string]ListenerStatus)
+
+	if !gateway.Valid {
```

Comment:
do we really need to have a special case here?

I think we can just rely on `conditions.DeduplicateConditions` - they will make sure Accepted False will override any Accepted True default conditions.

Reply:
Spoke offline. This case short-circuits if the gateway is invalid.